pkg/registry/persistentvolume/rest.go

```diff
@@ -0,0 +1,124 @@
+/*
+Copyright 2015 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package persistentvolume
+
+import (
+	"fmt"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api/errors"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api/validation"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/fields"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/labels"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/registry/generic"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/runtime"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/watch"
+)
+
+// REST implements the RESTStorage interface in terms of a PodRegistry.
+type REST struct {
+	registry generic.Registry
+}
+
+type RESTConfig struct {
+	Registry generic.Registry
+}
+
+// NewREST returns a new REST.
+func NewREST(registry generic.Registry) *REST {
+	return &REST{
+		registry: registry,
+	}
+}
+
+func (*REST) New() runtime.Object {
+	return &api.PersistentVolume{}
+}
+
+func (*REST) NewList() runtime.Object {
+	return &api.PersistentVolumeList{}
+}
+
+func (rs *REST) Create(ctx api.Context, obj runtime.Object) (runtime.Object, error) {
+	persistentvolume := obj.(*api.PersistentVolume)
+	api.FillObjectMetaSystemFields(ctx, &persistentvolume.ObjectMeta)
+	if errs := validation.ValidatePersistentVolume(persistentvolume); len(errs) > 0 {
+		return nil, errors.NewInvalid("persistentVolume", persistentvolume.Name, errs)
+	}
+
+	err := rs.registry.CreateWithName(ctx, persistentvolume.Name, persistentvolume)
+	if err != nil {
+		return nil, err
+	}
+
+	return rs.registry.Get(ctx, persistentvolume.Name)
+}
+
+func (rs *REST) Update(ctx api.Context, obj runtime.Object) (runtime.Object, bool, error) {
```

Comment:
So am I understanding this right that Update lets you modify Status and not just Spec right now?

Is there a plan to add a Status operation to enforce updates to just that section?